app/js/app/directives/equation_editor/EquationEditor.js

```diff
@@ -111,20 +117,41 @@ define(function(require) {
                     }
                 };
 
-                $rootScope.showEquationEditor = function(initialState, questionDoc) {
+                $rootScope.showEquationEditor = function(initialState, questionDoc, editorMode) {
 
                     return new Promise(function(resolve, reject) {
 
                         delete scope.symbolLibrary.customVars;
                         delete scope.symbolLibrary.customFunctions;
-                        if (questionDoc && questionDoc.availableSymbols) {
+                        delete scope.symbolLibrary.customChemicalSymbols;
+                        delete scope.symbolLibrary.customFunction;
+                        delete scope.symbolLibrary.augmentedOps;
+
+                        if (editorMode == "maths" && questionDoc && questionDoc.availableSymbols) {
                             var parsed = parseCustomSymbols(questionDoc.availableSymbols);
                             if (parsed.vars.length > 0) {
                                 scope.symbolLibrary.customVars = parsed.vars;
                             }
                             if (parsed.fns.length > 0) {
                                 scope.symbolLibrary.customFunctions = parsed.fns;
                             }
+                            var parsedOps = parseCustomOperations(questionDoc.availableSymbols);
```

Comment:
Is it not possible to make `parseCustomSymbols(...)`  also deal with operators too? That would be nice! make sure you avoid null pointers when returning `parsed.vars`, `parsed.fns` and the new `parsed.ops` though!

Do we need all these `console.dubug`s now?